qsimcirq_tests/qsimcirq_test.py

```diff
@@ -221,6 +221,70 @@ def test_matrix2_gate(self):
         result.state_vector(), cirq_result.state_vector())
 
 
+  def test_big_matrix_gates(self):
+    qubits = cirq.LineQubit.range(3)
+    # Toffoli gate as a matrix.
+    m = np.array([
```

Comment:
I think the Toffoli matrix here and below should be in the standard form. One thing is that this test tests only the first column of the matrix.

Reply:
Done. Also modified the circuit to ensure the Toffolis aren't no-ops.

I'm not certain what you mean by only testing the first column of the matrix - this test should execute the circuit (Toffolis included) and verify that the resulting state matches the result from Cirq.

Reply:
The starting state vector is [1, 0, 0, 0, 0, 0, 0, 0]. Only the first column of the matrix is relevant in the matrix-vector product in this case.

Reply:
Ah, I see. Added Hadamards on the control qubits for a more complete test of the matrix.

The main failure case we're concerned about here is being unable to decompose, but I agree it's better to fully test the matrix being used (e.g. if it was translated incorrectly somehow).